test/jdk/java/lang/instrument/GetObjectSizeIntrinsicsTest.java

```diff
@@ -264,25 +264,66 @@
  *                   -javaagent:basicAgent.jar GetObjectSizeIntrinsicsTest GetObjectSizeIntrinsicsTest
  */
 
+/*
+ * @test
+ * @summary Test for fInst.getObjectSize with large arrays
+ * @library /test/lib
+ * @requires vm.bits == 64
+ * @requires vm.debug
+ * @requires os.maxMemory >= 10G
+ *
+ * @build sun.hotspot.WhiteBox
+ * @run build GetObjectSizeIntrinsicsTest
+ * @run shell MakeJAR.sh basicAgent
+ *
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller sun.hotspot.WhiteBox
+ *
+ * @run main/othervm -Xmx8g
+ *                   -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure -XX:+WhiteBoxAPI -Xbootclasspath/a:.
+ *                   -XX:ObjectAlignmentInBytes=32
+ *                   -Xint
+ *                   -javaagent:basicAgent.jar GetObjectSizeIntrinsicsTest GetObjectSizeIntrinsicsTest large
+ *
+ * @run main/othervm -Xmx8g
+ *                   -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure -XX:+WhiteBoxAPI -Xbootclasspath/a:.
+ *                   -XX:ObjectAlignmentInBytes=32
+ *                   -Xbatch -XX:TieredStopAtLevel=1
+ *                   -javaagent:basicAgent.jar GetObjectSizeIntrinsicsTest GetObjectSizeIntrinsicsTest large
+ *
+ * @run main/othervm -Xmx8g
+ *                   -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure -XX:+WhiteBoxAPI -Xbootclasspath/a:.
+ *                   -XX:ObjectAlignmentInBytes=32
+ *                   -Xbatch -XX:-TieredCompilation
+ *                   -javaagent:basicAgent.jar GetObjectSizeIntrinsicsTest GetObjectSizeIntrinsicsTest large
+ */
+
 import java.util.*;
 
 import jdk.test.lib.Platform;
 import sun.hotspot.WhiteBox;
 
 public class GetObjectSizeIntrinsicsTest extends ASimpleInstrumentationTestCase {
 
-    static final Boolean compressedOops = WhiteBox.getWhiteBox().getBooleanVMFlag("UseCompressedOops");
-    static final int REF_SIZE = (compressedOops == null || compressedOops == true) ?  4 : 8;
+    static final Boolean COMPRESSED_OOPS = WhiteBox.getWhiteBox().getBooleanVMFlag("UseCompressedOops");
+    static final long REF_SIZE = (COMPRESSED_OOPS == null || COMPRESSED_OOPS == true) ? 4 : 8;
 
     static final Long align = WhiteBox.getWhiteBox().getIntxVMFlag("ObjectAlignmentInBytes");
     static final int OBJ_ALIGN = (align == null ? 8 : align.intValue());
 
-    public GetObjectSizeIntrinsicsTest(String name) {
+    static final int SMALL_ARRAY_SIZE = 1024;
+
+    // With int[] arrays, this overflows 4G boundary
+    static final int LARGE_ARRAY_SIZE = 1024*1024*1024 + 1024;
+
+    final String mode;
+
+    public GetObjectSizeIntrinsicsTest(String name, String mode) {
         super(name);
+        this.mode = mode;
     }
 
     public static void main(String[] args)throws Throwable {
-        new GetObjectSizeIntrinsicsTest(args[0]).runTest();
+        new GetObjectSizeIntrinsicsTest(args[0], (args.length >= 2 ? args[1] : "")).runTest();
```

Comment:
Shouldn't this be `args.length == 2`?

Reply:
`>= 2` is more future-proof, I think, as it checks that _at least_ two arguments exists, which allows to read `args[1]`. Accidental addition of third argument would not silently break the test.

Reply:
@plummercj, are you good with this explanation?

Reply:
@plummercj This does not look like a review-blocking comment, so I am going to proceed with integration, if there are no other comments.